src/package_json.rs

```diff
@@ -125,85 +64,125 @@ impl PackageJson {
+                        map.insert(dir.normalize_with(&key), map[&key].clone());
+                    }
+                    // Normalize the key if it is relative path "./relative"
+                    if key.starts_with(".") {
+                        if let Some(value) = map.remove(&key) {
+                            map.insert(dir.normalize_with(&key), value);
                         }
                     }
                 }
-                package_json.browser_fields.push(browser_field);
             }
+            package_json.browser_fields.push(browser_field);
         }
 
         // Dynamically create `exports`.
         for object_path in &options.exports_fields {
-            if let Some(exports) = Self::get_value_by_path(&package_json_value, object_path) {
-                let exports = ExportsField::deserialize(exports)?;
-                package_json.exports.push(exports);
+            if object_path.len() == 1 && object_path[0] == "exports" {
+                if let Some(exports) = &data.exports {
+                    package_json.exports.push(exports.to_owned());
+                }
+            } else if let Some(exports) = Self::get_value_by_path(&data.other_fields, object_path) {
+                package_json.exports.push(ImportExportField::deserialize(exports)?);
             }
         }
 
+        // Bubble up `imports`
+        if let Some(imports) = &data.imports {
+            package_json.imports = Box::new(imports.to_owned());
+        }
+
         package_json.path = path;
         package_json.realpath = realpath;
 
-        // Remove large fields that useless for pragmatic use, see [raw_json] API
-        if let Some(package_json_value) = package_json_value.as_object_mut() {
-            package_json_value.remove("description");
-            package_json_value.remove("keywords");
-            package_json_value.remove("scripts");
-            package_json_value.remove("dependencies");
-            package_json_value.remove("devDependencies");
-            package_json_value.remove("peerDependencies");
-            package_json_value.remove("optionalDependencies");
+        // Remove large fields that are useless for pragmatic use
+        if options.reduce_memory_usage {
+            data.scripts = None;
+            data.dependencies = None;
+            data.dependencies_meta = None;
+            data.dev_dependencies = None;
+            data.peer_dependencies = None;
+            data.peer_dependencies_meta = None;
+            data.bundle_dependencies = None;
+            data.optional_dependencies = None;
+            data.imports = None;
+            data.exports = None;
+            data.browser = None;
         }
 
-        package_json.raw_json = Arc::new(package_json_value);
+        package_json.data = data;
+
         Ok(package_json)
     }
 
     fn get_value_by_path<'a>(
-        package_json: &'a serde_json::Value,
+        fields: &'a FxHashMap<String, serde_json::Value>,
```

Comment:
The `other_fields` field in the struct: https://docs.rs/nodejs_package_json/0.1.1/nodejs_package_json/struct.PackageJson.html#structfield.other_fields